backend/app/controllers/spree/admin/products_controller.rb

```diff
@@ -30,6 +30,11 @@ def update
             param_attrs[:option_value_ids] = param_attrs[:option_value_ids].split(',')
           end
         end
+        if updating_variant_image_rules?
```

Comment:
I would rather see that done initially. Especially combining the rulesets, which will require a migration if done later to copy rules from one table to the merged one and fix up the links.